hashtable/hashtable.c

```diff
@@ -148,6 +148,7 @@ void table_insert(struct Table *table, const struct YASL_Object key, const struc
                                 return;
                         }
                 }
+		if ((unsigned int)i >= table->size) goto resize;
```

Comment:
I represents the number of times we've tried to hash so far, not anything related to the table size.

Also, we don't need this resize here because we've already resized in the cases where the hashtable was too full.